contracts/axelar-gas-service/src/test.rs

```diff
@@ -22,14 +26,87 @@ fn setup_env<'a>() -> (Env, Address, Address, AxelarGasServiceClient<'a>) {
 
     let client = AxelarGasServiceClient::new(&env, &contract_id);
     let gas_collector: Address = Address::generate(&env);
-    client.initialize(&gas_collector);
 
     (env, contract_id, gas_collector, client)
 }
 
+#[test]
+fn test_initialize() {
+    let (env, contract_id, gas_collector, client) = setup_env();
+    client.initialize(&gas_collector);
+
+    assert!(env.as_contract(&contract_id, || {
+        env.storage()
+            .instance()
+            .get::<DataKey, bool>(&DataKey::Initialized)
+            .unwrap()
+    }));
+
+    assert_eq!(
+        env.as_contract(&contract_id, || {
+            env.storage()
+                .instance()
+                .get::<DataKey, Address>(&DataKey::GasCollector)
+                .unwrap()
+        }),
+        gas_collector
+    );
+}
+
+#[test]
+fn fail_already_initialized() {
+    let (_env, _contract_id, gas_collector, client) = setup_env();
+    client.initialize(&gas_collector);
+
+    assert_contract_err!(
+        client.try_initialize(&gas_collector),
+        GasServiceError::AlreadyInitialized
+    );
+}
+
+#[test]
+fn fail_pay_gas_invalid_amount() {
+    let (env, contract_id, gas_collector, client) = setup_env();
+
+    let token_address: Address = env.register_stellar_asset_contract(Address::generate(&env));
+    let sender: Address = Address::generate(&env);
+    let gas_amount: i128 = 0;
+    let token = Token {
+        address: token_address.clone(),
+        amount: gas_amount,
+    };
+    let refund_address: Address = Address::generate(&env);
+    let payload = bytes!(&env, 0x1234);
+    let destination_chain: String = String::from_str(&env, "ethereum");
+    let destination_address: String =
+        String::from_str(&env, "0x4EFE356BEDeCC817cb89B4E9b796dB8bC188DC59");
+
+    let token_client = TokenClient::new(&env, &token_address);
+    StellarAssetClient::new(&env, &token_address).mint(&sender, &gas_amount);
+
+    let expiration_ledger = &env.ledger().sequence() + 200;
+
+    // approve token spend before invoking `pay_gas_for_contract_call`
+    token_client.approve(&sender, &contract_id, &gas_amount, &expiration_ledger);
+
+    assert_eq!(token_client.allowance(&sender, &contract_id), gas_amount);
+
+    assert_contract_err!(
+        client.try_pay_gas_for_contract_call(
+            &sender,
+            &destination_chain,
+            &destination_address,
+            &payload,
+            &refund_address,
+            &token,
+        ),
+        GasServiceError::InvalidAmount
+    );
+}
+
 #[test]
 fn pay_gas_for_contract_call() {
-    let (env, contract_id, _, client) = setup_env();
+    let (env, contract_id, gas_collector, client) = setup_env();
```

Comment:
same here, unused variable, keep it as `_`

Reply:
done